x/dao/keeper/delegation.go

```diff
@@ -139,14 +139,24 @@ func (k Keeper) computeDelegationsAndUndelegation(
 }
 
 // undelegateValidators undelegates the requested amount from the validators in the undelegations.
-func undelegateValidators(ctx sdk.Context, vals []stakingtypes.Validator, undelegations map[string]sdk.Dec, k Keeper, daoAddr sdk.AccAddress) error {
+func undelegateValidators(ctx sdk.Context, vals []stakingtypes.Validator, undelegations map[string]sdk.Int, k Keeper, daoAddr sdk.AccAddress) error {
 	for _, val := range vals {
 		valOper := val.GetOperator()
 		undelegationAmt, ok := undelegations[valOper.String()]
 		if !ok {
 			continue
 		}
-		undelegationShares := val.TokensFromShares(undelegationAmt)
+		if undelegationAmt.IsZero() {
+			continue
+		}
+		undelegationShares, err := val.SharesFromTokens(undelegationAmt)
```

Comment:
@cgdusek this is the reason of the issue. In case the shares rate is < 1 and you undelegate 1 token, it will lead to the undelegation of 0 shares, which lead to the error.

Reply:
Looks good. Thanks for tracking this down.